testing/e2e/skr-tester/pkg/keb/client.go

```diff
@@ -0,0 +1,341 @@
+		UserID:          config.UserID,
+		PlatformRegion:  config.PlatformRegion,
+	}
+}
+
+func (o *OAuthToken) GetToken(scopes string) (string, error) {
+	if o.Token != "" && time.Now().Before(o.Expiry) {
+		return o.Token, nil
+	}
+
+	data := fmt.Sprintf("grant_type=client_credentials&client_id=%s&client_secret=%s&scope=%s",
+		o.Credentials.ClientID, o.Credentials.ClientSecret, scopes)
+	req, err := http.NewRequest("POST", o.TokenURL, strings.NewReader(data))
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", errors.New("failed to get token")
+	}
+
+	var result map[string]interface{}
+	json.NewDecoder(resp.Body).Decode(&result)
+	o.Token = result["access_token"].(string)
+	o.Expiry = time.Now().Add(time.Duration(result["expires_in"].(float64)) * time.Second)
+
+	return o.Token, nil
+}
+
+func (c *KEBClient) BuildRequest(payload interface{}, endpoint, verb string) (*http.Request, error) {
+	token, err := c.Token.GetToken(SCOPES)
+	if err != nil {
+		return nil, err
+	}
+	platformRegion := c.GetPlatformRegion()
+	url := fmt.Sprintf("https://kyma-env-broker.%s/oauth/%sv2/%s", c.Host, platformRegion, endpoint)
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest(verb, url, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("X-Broker-API-Version", "2.14")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+	req.Header.Set("Content-Type", "application/json")
+	return req, nil
+}
+
+func (c *KEBClient) BuildRequestWithoutToken(payload interface{}, endpoint, verb string) (*http.Request, error) {
+	url := fmt.Sprintf("https://kyma-env-broker.%s/oauth/v2/%s", c.Host, endpoint)
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest(verb, url, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("X-Broker-API-Version", "2.14")
+	req.Header.Set("Content-Type", "application/json")
+	return req, nil
+}
+
+func (c *KEBClient) CallKEB(payload interface{}, endpoint, verb string) (map[string]interface{}, error) {
+	req, err := c.BuildRequest(payload, endpoint, verb)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusAccepted {
+		return nil, fmt.Errorf("error calling KEB: %s %s", resp.Status, resp.Status)
+	}
+
+	var result map[string]interface{}
+	json.NewDecoder(resp.Body).Decode(&result)
+	return result, nil
+}
+
+func (c *KEBClient) CallKEBWithoutToken(payload interface{}, endpoint, verb string) error {
```

Comment:
Should we use `KEB`? I am hesitant.
Then we have such awkward names (given the snakeCase convention).
Full name would be even more awkward `CallKymaEnvironmentBrokerWithoutToken`
What about `Broker`?

Reply:
We can discuss it with the team today.

Reply:
We decided to change KEB to broker and SKR to instance.